src/test/integrationSuiteSetup.ts

```diff
@@ -3,7 +3,19 @@
  * via `setupFilesAfterEnv`.
  */
 import { db } from '../database';
+import {
+  prepareDatabaseForCurrentWorker,
+  cleanupDatabaseForCurrentWorker,
+} from './harnessFunctions';
```

Comment:
It looks good. I am trying to imagine if we will have "int" tests that want to opt-out from database setup and teardown. In other words, despite the repetition, would it be worthwhile to include a `beforeEach` declaration for each `....int.test.ts` file in the `describe` block? It is clearer, I think, to have each integration test file have its own beforeEach/afterEach, but two counts against that approach are:
1. We have a suite/all-tests level `afterAll` call in `integrationSuiteSetup`, and
2. I am pretty sure all these tests will need the database (if not, it's a 50-500ms no-op).

Reply:
I agree with the concerns for what it's worth.

My ideal solution would be Jest just having a more robust way of defining categories of tests (e.g. having a `@usesDatabase` tag for the test, which isn't a thing but wouldn't it be nice).

Maybe we refactor to that down the line when we know more about our test suite?